### PR TITLE
stop sending update events when noop deleted of relationship properties occur

### PR DESCRIPTION
**Pull request by wheresrhys**
## Why?
Since adding rich relationships to the model edits made in Biz Ops admin to systems have been causing excess update events in the kinesis stream.

After digging around quite a bit I discovered that the form was posting e.g `dependencies: [{code: 'abc', 'resiliencePatterns': null}]` when no relationship properties were added. The API was unable to tell that writing `null` to a previously undefined property had no effect, and instead thought this was a substantive change, causing an unnecessary write to the database and several change events in the kinesis stream.
## What?
- adds a test for this edge case
- explicitly handles the case where the existing property is not defined and a null value is sent, and treats this as 'no difference'

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "concurrently": "^5.0.0",
     "css-loader": "^3.2.1",
     "cypress": "^3.8.2",
-    "fetch-mock": "^8.0.0",
+    "fetch-mock": "^9.9.0",
```
Comment (wheresrhys) [inline package.json:41]: also upgrades fetch-mock. Totally unrelated to the rest of the PR but I needed to test out a new release in a few projects and treecreeper is  a good 'n' complex one. May as well keep the upgrade now it's done

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 2]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
 			]);
 			expectUpdateEvent(childType, childCode, ['isChildOf']);
 		});
+
+		it('should not send extra UPDATE events when making noop property removal edit to rich relationship', async () => {
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:229]:
```suggestion
		it('should not send extra UPDATE events when removing rich relationship property that was already non-existent', async () => {
```

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 3]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
 			]);
 			expectUpdateEvent(childType, childCode, ['isChildOf']);
 		});
+
+		it('should not send extra UPDATE events when removing rich relationship property that was already non-existent', async () => {
+			const [main, child] = await Promise.all([
+				createMainNode(),
+				createNode(childType, { code: childCode }),
+			]);
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:233]: Create some fixture records - a `MainType` (`createMainNode() is a shorthand for `createNode('MainType', { code: mainCode })`... maybe that's a bit too DRY??

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 4]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+			await connectNodes(main, 'HAS_CURIOUS_CHILD', child);
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:234]: create a relationship between the fixtures

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 5]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+
+			const { status } = await patchHandler()(
+				getInput(
+					{
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:238]: getInput is a helper that return an object of the form `{code, type, body, metadata, query}`... i.e. the input for the rest handler functions

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 6]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+
+		it('should not send extra UPDATE events when removing rich relationship property that was already non-existent', async () => {
+			const [main, child] = await Promise.all([
+				createMainNode(),
+				createNode(childType, { code: childCode }),
+			]);
+			await connectNodes(main, 'HAS_CURIOUS_CHILD', child);
+
+			const { status } = await patchHandler()(
+				getInput(
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:237]: getInput is a helper that return an object of the form {code, type, body, metadata, query}... i.e. the input for the rest handler functions

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 7]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+					{
+						someString: 'some string',
+						curiousChild: {
+							code: childCode,
+							someString: null,
+						},
+					},
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:244]: The object above here is the `body` that gets sent. i.e., editing the `someString` property and basically writing the same relationship as already exists, including trying to delete a property on it which doesn't exist

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 8]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+					{ relationshipAction: 'replace', upsert: true },
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:245]: the query params for the request

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 9]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+				),
+			);
+
+			expect(status).toBe(200);
+			expect(emitSpy).toHaveBeenCalledTimes(1);
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:250]: this is a jest spy which keeps an eye on the event emitter built into the handler function

File: packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js [hunk 10]
```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
...
+			expectUpdateEvent(mainType, mainCode, ['someString']);
```
Comment (wheresrhys) [inline packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:251]: this helper is basically saying `expect(emitSpy).calledWith({type: mainType, code: mainCode, updatedProperties: ['someString']})`

Is it too DRY to have a helper for this?? I dunno. Before writing helpers like this and the one I mention a few comments above the tests were equally unreadable due to the masses of repeated, verbose object literals. At least with the helpers once you've learned them the tests are easier to scan. But open to discussion. Perhaps what's needed is some 'before you read the tests, know this...' documentation

Comment (i-like-robots) [reply packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js:251]: It's this bit I don't understand - not because it's been extracted into a helper but because it's not clear why the input (below) would result in this outcome. Are there separate tests for the diff algorithm?

```
{
    someString: 'some string',
    curiousChild: {
        code: childCode,
        someString: null,
    },
}
```
